src/server/lib/getRegistrationLocation.ts

```diff
@@ -0,0 +1,279 @@
+import { Request } from 'express';
+import { CountryCode } from '@guardian/libs';
+import { RegistrationLocation } from '../models/okta/User';
+
+/**
+ * The registrationLocation regions are derived from the first column (region_code)
+ * of the Data Design team regional classifications here:
+ * https://github.com/guardian/data-platform-models/blob/1f21699dd8a3609ee158e74daee79a58f4e7ebc1/dbt/data/seed_country.csv
+ * (Notes: ROW British Virgin Islands (VG) and EU Kosovo (XK) have been removed to validate type checking)
+ *
+ * The registrationLocation field is stored in Okta and can by changed by user via ManageMyAccount.
+ */
+export const getRegistrationLocation = (
+  req: Request,
+  cmpConsentedAll = false,
+): RegistrationLocation | undefined => {
+  const country: CountryCode | undefined = req.cookies['GU_geo_country'];
+
+  // registrationLocation must never be captured without full cmp consent
+  if (!cmpConsentedAll) return undefined;
+
+  if (country === 'GB') return 'United Kingdom';
+  if (Us.includes(country)) return 'United States';
+  if (country === 'AU') return 'Australia';
+  if (country === 'NZ') return 'New Zealand';
+  if (country === 'CA') return 'Canada';
+  if (Europe.includes(country)) return 'Europe';
+  if (Row.includes(country)) return 'Other';
+
+  return undefined;
+};
+
+const Us: (CountryCode | undefined)[] = ['US', 'AS', 'GU', 'MP', 'PR', 'VI'];
+
+// XK Kosovo removed
```

Comment:
👀 very minor in terms of code, very major in terms of political ramifications

I'd like to understand better why `VG` and `XK` have been removed, as they seem like a strange omission, and `removed to validate type checking` doesn't make much sense to me.

Reply:
This interested me so I went digging!

Looks like this is the origin type for `CountryCode`: https://github.com/guardian/csnx/blob/main/libs/%40guardian/libs/src/countries/%40types/CountryCode.ts

It derives from [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). That standard _does_ have `VG` for Virgin Islands, so that's a straight up omission on the part of the Guardian libs type. It _doesn't_ have Kosovo, which I believe is because it's not listed in any of the UN sources which make up the list, which is ultimately because it's only partially recognised.

Reply:
> XK is a "user assigned" ISO 3166 code not designated by the standard, but used by the [European Commission](https://en.wikipedia.org/wiki/European_Commission), Switzerland, the [Deutsche Bundesbank](https://en.wikipedia.org/wiki/Deutsche_Bundesbank) and other organisations. However, [ISO 3166-2:RS-KM](https://en.wikipedia.org/wiki/ISO_3166-2:RS) remains in use.

Reply:
I see no reason why we wouldn't want to include `XK` it in our libs - we're not trying to adhere to strict ISO 3166 as far as I'm aware!

Reply:
At least I don't think we should mention that we explicitly left `XK` out, as that may not reflect well on the whole organisation.

Reply:
I guess this thread is the context 🤷🏾

Reply:
I've raised a PR to add VG in the library but will leave XK off for now unless that team chooses to add it to the library as unsure if it needs strict ISO3166 adherence or not -  https://github.com/guardian/csnx/pull/147